### PR TITLE
[DO NOT MERGE]: S2 delegates

### PR DESCRIPTION
**Pull request by TarunAdobe**
## Description

Introducing a new delegate based pattern to manage specific controllers based on the current spectrum version.

## Related issue(s)

-

## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [ ] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [ ] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by Westbrook** [comment]
If you catch this up to main, do the no change diffs go away?

File: packages/button/src/ButtonBase.ts [hunk 1]
```diff
@@ -32,6 +32,10 @@ import buttonStyles from './button-base.css.js';
 export class ButtonBase extends ObserveSlotText(LikeAnchor(Focusable), '', [
     'sp-overlay,sp-tooltip',
 ]) {
+    spectrumConfig = {
+        downstate: true,
```
Comment (Westbrook) [inline packages/button/src/ButtonBase.ts:36]: We likely want the value here not to be a boolean but an array of values that could include the current "system", e.g.:
```suggestion
        downstate: ['spectrum2'],
```

Comment (Westbrook) [reply packages/button/src/ButtonBase.ts:36]: This way, some functionality can be delivered a la:
```
    downstate: ['spectrum1', 'spectrum2']
```

We should work to include these values an enums, to simplify sharing and consumption.

Comment (TarunAdobe) [reply packages/button/src/ButtonBase.ts:36]: Great idea! I'll update it accordingly

**Comment by TarunAdobe** [comment]
> If you catch this up to main, do the no change diffs go away?

No I don't think that'd be the case as I've manually created fake stylesheets for S2 theme so this PR isn't mergeable

**Comment by Rajdeepc** [comment]
> If you catch this up to main, do the no change diffs go away?

I think this must be due to `git add .` . There are many stale files went in! 
cc @TarunAdobe

**Comment by Rajdeepc** [comment]
@TarunAdobe Can we close this?

**Comment by TarunAdobe** [comment]
Closing this as we have decided to make separate PRs for individual feature

**Pull request closed without merging**